core/src/test/java/hudson/util/io/TarArchiverTest.java

```diff
@@ -83,19 +83,19 @@
             // extract via the tar command
             run(e, "tar", "xvpf", tar.getAbsolutePath());
 
-            assertEquals(0100755,e.child("a.txt").mode());
+            assertEquals(0755,e.child("a.txt").mode());
             assertEquals(dirMode,e.child("subdir").mode());
-            assertEquals(0100644,e.child("subdir/b.txt").mode());
+            assertEquals(0644,e.child("subdir/b.txt").mode());
```

Comment:
We could try to set the file type bits in the new `IOUtils#mode`, but we would only be able to reliably set the type for regular files, symbolic links, and directories, so it seems better to only return the basic permissions and document that fact.